Dockerfile

```diff
@@ -1,23 +1,28 @@
-# Build the manager binary
-FROM golang:1.13.3 as builder
+FROM golang:1.14-alpine as builder
+
+ARG TARGETPLATFORM
+
+ENV GO111MODULE=on \
+    CGO_ENABLED=0
+
+RUN apk add --no-cache git
 
-# Copy in the go src
 WORKDIR /go/src/github.com/open-policy-agent/gatekeeper
-COPY pkg/    pkg/
-COPY third_party/ third_party/
-COPY vendor/ vendor/
-COPY main.go main.go
-COPY api/ api/
-COPY go.mod .
 
-# Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -mod vendor -a -o manager main.go
+RUN export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) && \
+    export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2) && \
+    GOARM=$(echo ${TARGETPLATFORM} | cut -d / -f3); export GOARM=${GOARM:1} && \
+    git clone https://github.com/open-policy-agent/gatekeeper.git . && \
```

Comment:
We should keep the above COPY code instead of cloning from GitHub so both for hermeticity and so that we can build from changes not yet checked into master during development.

Reply:
Fixed - Updated .dockerignore to effectively COPY only the required files that were previously explicitly mentioned, whilst keeping it to a single COPY layer.